client/my-sites/post-type-list/post-type-site-info/index.jsx

```diff
@@ -0,0 +1,49 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import { isEnabled } from 'config';
+import { getPost } from 'state/posts/selectors';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import { getSite, getSiteTitle } from 'state/sites/selectors';
+import SiteIcon from 'blocks/site-icon';
+
+function PostTypeSiteInfo( { site, siteTitle, isAllSitesModeSelected } ) {
+	if ( ! isEnabled( 'posts/post-type-list' ) || ! site || ! isAllSitesModeSelected ) {
+		return null;
+	}
```

Comment:
As noted above, I think we can avoid depending on `isEnabled` and `isAllSitesModeSelected` here, by allowing the parent component to decide whether to render this component.

`if ( ! site )` - is this condition possible / likely / easy to reproduce?  When that happens, should this component return a placeholder instead?

Reply:
I just pushed a commit to allow the parent to decide about the visibility of the child component.

`if ( ! site )` is `true` while the Blog Posts page is being loaded in All Sites mode. For me, that condition is hit 15-20 times before the site becomes known. So it's possible, and easy to reproduce. The post item card already looks like a placeholder while the post list is being loaded so visually I don't see any issue without rendering a placeholder here explicitly.